apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts

```diff
@@ -0,0 +1,688 @@
+        assessmentInstanceIdColumn,
+        {
+          field: 'group_name',
+          title: 'Name',
+          visible: false,
+          sortable: true,
+          class: 'align-middle',
+          switchable: true,
+        },
+        {
+          field: 'uid_list',
+          title: 'Group Members',
+          sortable: true,
+          class: 'text-center align-middle text-wrap',
+          formatter: listFormatter,
+          switchable: true,
+        },
+        {
+          field: 'user_name_list',
+          title: 'Group Member Name',
+          sortable: true,
+          visible: false,
+          class: 'text-center align-middle text-wrap',
+          formatter: listFormatter,
+          switchable: true,
+        },
+        {
+          field: 'group_roles',
+          title: `Roles <button class="btn btn-xs" type="button" title="Show roles help" data-toggle="modal" data-target="#role-help"><i class="bi-question-circle-fill" aria-hidden="true"></i></button>`,
+          sortable: true,
+          class: 'text-center align-middle text-wrap',
+          formatter: uniqueListFormatter,
+          switchable: true,
+        },
+        ...nonspecificColumns,
+        actionButton,
+      ];
+    } else {
+      return [
+        assessmentInstanceIdColumn,
+        {
+          field: 'uid',
+          title: 'UID',
+          visible: false,
+          sortable: true,
+          class: 'align-middle text-nowrap',
+          switchable: true,
+        },
+        {
+          field: 'name',
+          title: 'Name',
+          sortable: true,
+          class: 'align-middle text-nowrap',
+          switchable: true,
+        },
+        {
+          field: 'role',
+          title:
+            'Role <button class="btn btn-xs" type="button" title="Show roles help" data-toggle="modal" data-target="#role-help"><i class="bi-question-circle-fill" aria-hidden="true"></i></button>',
+          sortable: true,
+          class: 'text-center align-middle text-nowrap',
+          switchable: true,
+        },
+        ...nonspecificColumns,
+        {
+          field: 'client_fingerprint_id_change_count',
+          title:
+            'Fingerprint Changes <button class="btn btn-xs", type="button" title="Show figerprint changes help" data-toggle="modal" data-target="#fingerprint-changes-help"><i class="bi-question-circle-fill" aria-hidden="true"></i></button>',
+          class: 'text-center align-middle',
+          switchable: true,
+        },
+        actionButton,
+      ];
+    }
+  }
+
+  function refreshTable() {
+    bsTable.bootstrapTable('refresh', { silent: true });
+  }
+
+  window.popoverSubmitViaAjax = function (e: any, popover) {
+    e.preventDefault();
+    $.post(
+      $(e.target).attr('action') ?? '',
+      $(e.target).serialize(),
+      function () {
+        refreshTable();
+      },
+      'json',
+    );
+    $(popover).popover('hide');
+  };
+
+  function timeLimitEditPopoverContent(this: HTMLElement) {
```

Comment:
It'd be great to use an `html` tagged-template literal to construct all the HTML here, just like if we were on the server!

Reply:
Updated to html template literals here and elsewhere in [306ccbe](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/306ccbe9c0ebd4481747d7e7ddce945db6fff328).